pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -683,7 +667,47 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongos(cr *api.PerconaServerMon
 	if deplSpec.Template.Annotations == nil {
 		deplSpec.Template.Annotations = make(map[string]string)
 	}
-	for k, v := range annotations {
+
+	if len(secretAnnotations) == 0 {
+		// we shoud check if sfs secrets were changes
+		// if yes - update mongos secret hash also
```

Comment:
all these logic restart mongos on any user secret change.
but in fact, we need restart only if Cluster Admin or Monitor users were changed

Reply:
If we update users all ReplicaSets will be restarted. When we restart RS we also have to restart mongos.